sync/evmdownloader.go

```diff
@@ -104,6 +108,18 @@ func (d *EVMDownloader) setStopDownloaderOnIterationN(iteration int) {
 	d.stopDownloaderOnIterationN = iteration
 }
 
+// GetRuntimeData returns the runtime data: chainID + addresses to query
+func (d *EVMDownloader) GetRuntimeData(ctx context.Context) (RuntimeData, error) {
+	chainID, err := d.GetChainID(ctx)
+	if err != nil {
+		return RuntimeData{}, err
+	}
+	return RuntimeData{
+		ChainID:   chainID,
+		Addresses: d.adressessToQuery,
+	}, nil
```

Comment:
Related to the other comment:
```suggestion
	chainID, err := d.ChainID(ctx)
	if err != nil {
		return RuntimeData{}, err
	}
	return RuntimeData{
		ChainID:   chainID.Uint64,
		Addresses: d.adressessToQuery,
	}, nil
```